kbase-extension/static/kbase/js/common/jobCommChannel.js

```diff
@@ -64,8 +64,12 @@ define([
                       /* no op */
                   };
 
-            this.messageQueue = [];
             this.ERROR_COMM_CHANNEL_NOT_INIT = 'Comm channel not initialized, not sending message.';
+
+            this.messageQueue = [];
+
+            // maintain a mapping of job IDs and dispatch information
+            this._jobMapping = {};
```

Comment:
I see this getting set up and populated throughout the changes, but what's it used for? Or will that be in the next stage to go from `job_id` to its batch parent? and from batch ids to their children?

Reply:
The next step is using this mapping to group together messages with info about jobs in the same batch and send them out on the channel for the batch parent. This will require changes to both the job comm channel and the message handlers that deal with the data in frontend widgets like the job status table.